source/common/network/connection_impl.cc

```diff
@@ -262,6 +290,19 @@ void ConnectionImpl::closeSocket(ConnectionEvent close_type) {
 
   connection_stats_.reset();
 
+  if (enable_rst_detect_send_ && (detected_close_type_ == DetectedCloseType::RemoteReset ||
+                                  detected_close_type_ == DetectedCloseType::LocalReset)) {
+    const bool ok = Network::Socket::applyOptions(
+        Network::SocketOptionFactory::buildZeroSoLingerOptions(), *socket_,
+        envoy::config::core::v3::SocketOption::STATE_LISTENING);
+    if (!ok) {
+      ENVOY_CONN_LOG(warn, "rst setting so_linger=0 socket options failed", *this);
```

Comment:
This can be very spammy, in can it is failing consistently. Using ENVOY_LOG_EVERY_POW_2 with `error` might be a better choice here to avoid log spam.

Also I do not think the log message is needed for the success case.

Reply:
Thanks! done.